src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperRequiredAttributeDescriptorComparer.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.Compilation.TagHelpers
+{
+    public class TagHelperRequiredAttributeDescriptorComparer : IEqualityComparer<TagHelperRequiredAttributeDescriptor>
+    {
+        public static readonly TagHelperRequiredAttributeDescriptorComparer Default = new TagHelperRequiredAttributeDescriptorComparer();
+
+        protected TagHelperRequiredAttributeDescriptorComparer()
+        {
+        }
+
+        public virtual bool Equals(TagHelperRequiredAttributeDescriptor descriptorX, TagHelperRequiredAttributeDescriptor descriptorY)
+        {
+            if (descriptorX == descriptorY)
+            {
+                return true;
+            }
+
+            return descriptorX != null &&
+                descriptorX.Selector == descriptorY.Selector &&
+                descriptorX.IsCSSSelector == descriptorY.IsCSSSelector &&
+                string.Equals(descriptorX.Name, descriptorY.Name, StringComparison.OrdinalIgnoreCase) &&
+                string.Equals(descriptorX.Value, descriptorY.Value, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public virtual int GetHashCode(TagHelperRequiredAttributeDescriptor descriptor)
+        {
+            var hashCodeCombiner = HashCodeCombiner.Start();
+            hashCodeCombiner.Add(descriptor.Selector);
+            hashCodeCombiner.Add(descriptor.IsCSSSelector);
+            hashCodeCombiner.Add(descriptor.Name, StringComparer.OrdinalIgnoreCase);
+            hashCodeCombiner.Add(descriptor.Value, StringComparer.Ordinal);
```

Comment:
`StringComparer` here should alight with that used [above](https://github.com/aspnet/Razor/pull/692/files#diff-2c1e885030894d93397bf7811919cbedR29).